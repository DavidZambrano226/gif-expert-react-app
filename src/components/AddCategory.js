import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    
    const [inputValue, setInputValue] = useState('');
    
    const handlerChange = (e) => {
        setInputValue(e.target.value)
    }

    const handlerSubmit = (e) => {
        e.preventDefault();
        
        if (inputValue.trim().length > 2 ) {
            setCategories( category => [inputValue, ...category] );
            setInputValue('');
        }

    }
    
    return (
        <form onSubmit={handlerSubmit}>
            <input 
                type="text" 
                value = {inputValue}
                onChange={ handlerChange } 
                
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
