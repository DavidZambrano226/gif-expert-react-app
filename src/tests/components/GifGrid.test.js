import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGif } from '../../hooks/useFetchGif';
jest.mock('../../hooks/useFetchGif');

describe('Test form GifGrid component', () => {
    const category = 'Cats';
    
    test('should make match with the snapshot', () => {

        useFetchGif.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={category} /> );
        expect( wrapper ).toMatchSnapshot();
    })

    test('should show items when imges are loaded', () => {
        
        const gifs = [
            {
                id: 'ABCD',
                url: 'http://localhost/img.jpg',
                title: 'Test image'
            },
            {
                id: 'ABCDEF',
                url: 'http://localhost/img.jpg',
                title: 'Test image Two'
            }
        ];
        
        useFetchGif.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category} /> );
        
        expect( wrapper.find('p').exists() ).toBeFalsy();
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );


    })
    
    
})
