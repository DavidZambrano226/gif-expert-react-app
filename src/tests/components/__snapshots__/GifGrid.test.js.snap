// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test form GifGrid component should make match with the snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Cats
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
